src/ros_commo.py

```diff
@@ -376,15 +402,85 @@ def behavior_switch_callback(self, data):
 	# Data is a bit-flag that enables/disables publication of messages.
 	def behavior_control_callback(self, data):
 		self.control_mode = data.data
-
+	
+	def face_loc_cb(self, data):
+		#print "@@@@@@@@@@ In face cb"
+		self.lock.acquire()
+		try:
+			if len(self.sc_str_set)<1:
+				self.face_array=data.faces
+				self.sc_str_set="do"
+		finally:
+			self.lock.release()
+		#for face in data.faces:
+			#fac="(map-ato \"faces\" (NumberNode \""+str(face.id)+"\") "+str(face.point.x)+" "+str(face.point.y)+" "+str(face.point.z)+")"
+			#scheme_eval(self.atomspace,fac)
+
+	def thr(self):
+		self.atomspace = AtomSpace()#scheme_eval_as('(cog-atomspace)')
+		lpth="/home/mandeep/hr/opencog/ros-behavior-scripting/src/time-map.scm"
+		scheme_eval(self.atomspace, "(load \""+lpth+"\")")
+		#print scheme_eval(self.atomspace,"(+ 2 3)")
+		#scheme_eval(self.atomspace, "(map-ato \"faces\" (NumberNode \"2.0\") 1 2 3)")
+		rate=rospy.Rate(30)
+		while not rospy.is_shutdown():
+			rate.sleep()
+			self.lock.acquire()
```

Comment:
why do you need to do this in a new thread? Why can't you just do this in the main loop?